### PR TITLE
Fix Antithesis NullPointerException in Evm

### PR DESCRIPTION
**Pull request by Demuirgos**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- Fix Antithesis NullException 

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs [hunk 1]
```diff
@@ -283,6 +283,13 @@ protected virtual bool ValidateStatic(Transaction tx, BlockHeader header, IRelea
                     QuickFail(tx, header, spec, tracer, "block gas limit exceeded");
                     return false;
                 }
+
+                if (tx.IsContractCreation && (tx.Data is null || tx.DataLength == 0))
+                {
+                    TraceLogInvalidTx(tx, $"INVALID_CREATE_TRANSACTION_MISSING_INITCODE");
+                    QuickFail(tx, header, spec, tracer, "contract creation without Input initcode");
+                    return false;
```
[the remark this answers is not in the document]
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:291]: not a spec. resolving this:
https://hackmd.io/@opBRCgEeSP6XKr0nO99rbQ/Sy8I87wyp#Nethermind

[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:291]: transaction processing is part of consensus - it should have spec, we cannot just do as we please.

[the remark this answers is not in the document]
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:291]: okay so maybe we can put this validation at JsonRpc level or the blockchain bridge?

[the remark this answers is not in the document]
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:291]: you are absolutely right, we are failing ethereum tests with this. damn

**Review by smartprogrammer93 — CHANGES_REQUESTED** [review]
need to move the validation up to JsonRPC level maybe

**Comment by smartprogrammer93** [comment]
@LukaszRozmej  we checked geth behavior and copied the same

**Review by smartprogrammer93 — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

**Review by benaadams — APPROVED** [review]

**Pull request merged**